group_project_v2/mixins.py

```diff
@@ -145,43 +340,25 @@ def is_group_member(self):
         """
         return self.user_id in [u.id for u in self.workgroup.users]
 
-    @staticmethod
-    def _confirm_outsider_allowed(project_api, user_id, course_id):
-        granted_roles = {r["role"] for r in project_api.get_user_roles_for_course(user_id, course_id)}
-        allowed_roles = set(ALLOWED_OUTSIDER_ROLES)
-
-        return bool(allowed_roles & granted_roles)
-
     @property
     def workgroup(self):
         """
         :rtype: WorkgroupDetails
         """
-        workgroup = self._get_workgroup(self.project_api, self.user_id, self.course_id)
-        return workgroup if workgroup else self.FALLBACK_WORKGROUP
-
-    @staticmethod
-    @memoize_with_expiration()
-    def _get_workgroup(project_api, user_id, course_id):
-        """
-        :rtype: WorkgroupDetails
-        """
         try:
-            user_prefs = UserAwareXBlockMixin._user_preferences(project_api, user_id)
-
+            user_prefs = UserAwareXBlockMixin._user_preferences(self.project_api, self.user_id)
             if UserAwareXBlockMixin.TA_REVIEW_KEY in user_prefs:
-                if not WorkgroupAwareXBlockMixin._confirm_outsider_allowed(project_api, user_id, course_id):
-                    raise OutsiderDisallowedError("User does not have an allowed role")
-                result = project_api.get_workgroup_by_id(user_prefs[UserAwareXBlockMixin.TA_REVIEW_KEY])
+                self.check_ta_access(self.user_id, self.course_id)
+                return self.project_api.get_workgroup_by_id(
+                    user_prefs[UserAwareXBlockMixin.TA_REVIEW_KEY])
```

Comment:
@jbzdak nit: I believe the convention is to have a closing bracket on the new line in such cases. Or at least that's a convention I was sticking to in this repo :)